backend/src/main/java/com/htecgroup/skynest/service/impl/InviteServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.htecgroup.skynest.service.impl;
+
+import com.htecgroup.skynest.annotation.EmailNotInUse;
+import com.htecgroup.skynest.model.dto.LoggedUserDto;
+import com.htecgroup.skynest.model.email.Email;
+import com.htecgroup.skynest.service.CurrentUserService;
+import com.htecgroup.skynest.service.EmailService;
+import com.htecgroup.skynest.service.InviteService;
+import com.htecgroup.skynest.util.EmailUtil;
+import com.htecgroup.skynest.util.JwtUtils;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+
+@Service
+@Log4j2
+@Validated
+@AllArgsConstructor
+public class InviteServiceImpl implements InviteService {
+
+  private final CurrentUserService currentUserService;
+  private final EmailService emailService;
+
+  @Override
+  public void sendRegistrationInvite(@EmailNotInUse String newUserEmail) {
+
+    LoggedUserDto loggedUserDto = currentUserService.getLoggedUser();
+
+    String token =
+        JwtUtils.generateRegistrationInviteToken(
+            loggedUserDto.getName(), loggedUserDto.getCompany().getName());
```

Comment:
shouldn't the new user's email be in the token?

Reply:
I literally can't comprehend what I was doing. In the function definition I game email and company name as arguments. Changing to email and company name.

Reply:
I think company emails aren't unique right now

you could add a script for that

Reply:
I meant personal emails, and I am thinking of making the company names unique too.